### ISSUE TITLE
FASTLED not working with ESP8266 Node MCU

### ISSUE THREAD

**Issue by samiraa78** [FastLED/FastLED#542]
Hi!

I am trying to use ESP8266 to  connect and control my LED from Hassio, the code i am using uses FASTLED however for some reason none of the demos of FAST LED or the code work, sometimes the first light lits and sometimes the light led lits thats all happens

Adifruit and all other library work fine any idea if i need to enter specific code for Node MCU

I am using WS2812B LED lights

**Comment by xkuyax** [comment]
Have you got a level shifter? The nodemcu has only a logic level of 3.3 Volts and the leds have a minimum logic level of 3.5V, it might work (it works for me), but it might not work for every led strip/controller combination.

**Comment by coryking** [comment]
Have you tried different pins?

**Comment by tonton81** [comment]
use pins 5 or 14
i did a demo using pin 14 of this github source
Tony

**Comment by tonton81** [comment]
and no level shifter either, 144 leds strip runs very well!

**Comment by dwc56** [comment]
did you get it working because i am trying do do the same thing?

**Comment by samiraa78** [comment]
Yes i dot it working in the code you need to change the pin address if you are mentioning D4 it wont work however if you only 4 it will work try doing that after couple of days of head banging this simple trick worked for me

**Comment by Muplex** [comment]
I've got the same issue. Neopixel works with adafruit library but not with fastled. How did you solve it?

**Comment by samiraa78** [comment]
What is the exact Pin number you have mentioned in the code if the pin is D4 and you have mentioned D4 trying changing it to 4 and see if it works

**Comment by Muplex** [comment]
indeed it works!!!! Thank you very much!!!!

I want to explain the situation for future readers;
In my sketch I was using buttons + RGB Neopixel 16 dots ring.

The confusion comes out because with FastLed.H different object initialization (in these case the neopixel) in witch I had to use the number print on the nodemcu (D0 = 0; D1 = 1; ...... FOR FASTLED RELATED OBJ) BUT to initialize the buttons I had to map the  pins as known (D0 = 16; D1 = 5; .......... FOR THE BUTTONS)...

**Comment by brendanmatkin** [comment]
As mentioned above NodeMCU re-maps the ESP8266 pins so they all line up nicely on the breakout. This image makes it pretty clear which is which (so if you want GPIO4 you can use D2, or GPIO3 is D9 (shown as RX on this picture)).

![](http://www.electronicwings.com/public/images/user_images/images/NodeMCU/NodeMCU%20Basics%20using%20Arduino%20IDE/NodeMCU%20GPIO/NodeMCU%20GPIOs.png)

**Comment by ArcadeNut** [comment]
I have the same issue.  I tested the adafruit Neopixel and it works with my WS2812B strip.  I'm using a Wemos D1 R1 board.  The strip turns on but erratic flashing.  i've use D5 or 15 and D4 or 4 no luck.  Its not a power issue as I'm power with a 5v 60a power supply and only 60LEDs.  I have a 470ohm resistor on the data line.

**Comment by TheOnlyJoey** [comment]
> I have the same issue. I tested the adafruit Neopixel and it works with my WS2812B strip. I'm using a Wemos D1 R1 board. The strip turns on but erratic flashing. i've use D5 or 15 and D4 or 4 no luck. Its not a power issue as I'm power with a 5v 60a power supply and only 60LEDs. I have a 470ohm resistor on the data line.

Sorry if i am bumping a way to old issue.
If you are feeding the neopixel with a 5v line, your trigger value will be 3.5v, which is 0.2v to high for the gpio levels of your board (with a max of 3.3v).

**Comment by shawheenattar** [comment]
Also sorry for bumping this but I'm also experiencing a similar issue. I got a level shifter to try to fix the problem, but it still isn't working with FastLED, although it does work with NeoPixel library without the level shifter. I've attached a pic of my wiring, so does anything look wrong? Would really appreciate the help. I've been banging my head on this for a few days now.

![IMG_1625](https://user-images.githubusercontent.com/12953284/59404163-1aabe200-8d6b-11e9-9155-fe106a74aca8.JPG)
